api/products/put.js

```diff
@@ -1,24 +1,50 @@
-import * as product from '../../db/tables/product.js'
-import * as helpers from '../../helpers/product.js'
-import getResponse from '../../helpers/response'
-
-
-export function updateProductById(req, res) {
-    let products = product.getProducts()
-
-    if(!helpers.isProductExists(products, req.params.productId)) {
-        return res.status(400).json({response: getResponse(false, "update",
-            "Element with id " + req.params.productId + " does not exist", "user")})
+import * as responseHelpers from '../../helpers/response'
+import {
+    OPERATION_TYPES,
+    ERRORS_DESCRIPTIONS
+} from '../constants'
+import {
+    updateProductByIdQuery,
+    getProductByIdQuery
+} from '../../sql-queries'
+import {
+    oneOrNone,
+    update
+} from '../../db'
+import { PRODUCTS } from '../../constants'
+import _ from 'lodash'
+import { isValuesValid } from '../../helpers/product'
+
+export async function updateProductById(req, res) {
+    let newValues = {}
+    _.forEach(PRODUCTS.COLUMNS, (value) => {
+        if (!(value === PRODUCTS.COLUMNS.ID) && req.body[value] ) {
+            newValues[value] = req.body[value]
+        }
+    })
+
+    try {
+        isValuesValid(newValues)
```

Comment:
Helpers must not throw exception. Leave //TODO: comment for it and let's discuss it later.